### PR TITLE
Docker build

### PR DESCRIPTION
**Pull request by warniaha**
https://github.com/zalsader/ceph/issues/72
https://github.com/zalsader/ceph/issues/65

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Review by zalsader — APPROVED** [review]

File: src/daos/build/install-single-host-dgw.sh [hunk 1]
```diff
@@ -0,0 +1,272 @@
+#!/bin/bash
+
+# fail on errors
+set -e
+set -x
+
+function set_boolean()
```
Comment (zalsader) [inline src/daos/build/install-single-host-dgw.sh:7]: This was moved to a file

File: src/daos/isRadosgwRunning.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
```
Comment (zalsader) [inline src/daos/isRadosgwRunning.sh:1]: It's better to use the same naming conventions, i.e. `is_radosgw_running`

File: src/daos/color_output.sh [hunk 3]
```diff
@@ -0,0 +1,140 @@
+#!/bin/bash
```
Comment (zalsader) [inline src/daos/color_output.sh:1]: These loose scripts could be grouped into directories: daos-related scripts, radosgw stuff, utils

File: src/daos/build/install-single-host-dgw.sh (hunk 1 above)
Comment (warniaha) [reply src/daos/build/install-single-host-dgw.sh:7]: you're right.  However in this specific case downloading it from the branch would be needed before setting the options for the repo & branch.  I did think about this one.  I'll change this one after the squash so it exists in the base branch before it gets used.

File: src/daos/color_output.sh (hunk 3 above)
Comment (warniaha) [reply src/daos/color_output.sh:1]: I'll do that after the squash

File: src/daos/isRadosgwRunning.sh (hunk 2 above)
Comment (warniaha) [reply src/daos/isRadosgwRunning.sh:1]: I'll do that after the squash

**Pull request merged**